packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -51,8 +57,42 @@ const defaultProps = {
 
 // eslint-disable-next-line react/prefer-stateless-function
 class DatePickerInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  }
+
+  handleOnButtonClick() {
+    // If the containing component (e.g. modal) has the focus trapped, the focus needs to be released so that the picker can get focus.
+    if (this.props.app && this.props.app.releaseFocus) {
+      this.props.app.releaseFocus();
```

Comment:
Looks to me like the naming seems backward. I would imagine that the picker component would be "requesting" focus for itself from the delegate when it is attempting to display it's FocusTrap, and "releasing" focus when it's hiding it's display.

I would imagine the request/release